tools/src/icon4pytools/py2fgen/wrappers/common.py

```diff
@@ -78,8 +82,8 @@ def construct_icon_grid(
     e2c2e = adjust_fortran_indices(e2c2e, offset)
 
     # stacked arrays
-    c2e2c0 = xp.column_stack((xp.asarray(range(c2e2c.shape[0])), c2e2c))
-    e2c2e0 = xp.column_stack((xp.asarray(range(e2c2e.shape[0])), e2c2e))
+    c2e2c0 = xp.column_stack((xp.asarray(range(c2e2c.shape[0])), c2e2c.ndarray))
```

Comment:
the connectvities that are used here are `xp.ndarray` not fields.You don need the extra .ndarray.
```suggestion
    c2e2c0 = xp.column_stack((xp.asarray(range(c2e2c.shape[0])), c2e2c))
```